### PR TITLE
Close connection when handshake timeout occurs

### LINKED ISSUE #483: not close udp conn when dtls handshake failed？
### Your environment.
  - Version: *2.1.5*
  - Browser: *include version*
  - Other Information - *stacktraces, related issues, suggestions how to fix, links for us to have context*

### What did you do?

when client connect to dtls service, if psk callback return error, handshake also return error but not return conn. I see the following code in conn.go:887, psk error not in fatal and close, so handshake function is close, but udp conn is running, how to close udp conn？
```
if e != nil {
	if e.IsFatalOrCloseNotify() {
		_ = c.close(false) //nolint:contextcheck
	}
}
```

### PR DESCRIPTION
**Pull request by jkralik**
Fixes #483

### CODE REVIEW

**Comment by jkralik** [comment]
@at-wat @Sean-Der Hi, sorry for interrupting, but is there something that needs to be done to review this PR? Thx

File: conn.go [hunk 1]
```diff
@@ -941,13 +947,18 @@ func (c *Conn) close(byUser bool) error {
...
+	isClosed := c.isConnectionClosed()
 	c.closed.Close()
 	c.closeLock.Unlock()
 
 	if closedByUser {
 		return ErrConnClosed
 	}
 
+	if isClosed {
+		return nil
```
Comment (at-wat) [inline conn.go:959]: `crypto/tls.Conn` (based on `net.TCPConn`) seems to return `use of closed network connection` error if closed twice.

Comment (at-wat) [reply conn.go:959]: Never mind.
In the current implementation, first `Close()` after timeout returns `nil` by this condition and second `Close()` returns `ErrConnClosed` by the above condition. It matches `crypto/tls.Conn`'s behavior.

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**